libs/execution_context/ebpf_core.c

```diff
@@ -1208,9 +1208,17 @@ ebpf_core_get_pinned_object(_In_ const cxplat_utf8_string_t* path, _Out_ ebpf_ha
 
     retval = ebpf_handle_create(handle, (ebpf_base_object_t*)object);
 
+    if (retval == EBPF_SUCCESS) {
+        EBPF_LOG_MESSAGE_UINT64(
+            EBPF_TRACELOG_LEVEL_ERROR,
+            EBPF_TRACELOG_KEYWORD_CORE,
+            "ebpf_core_get_pinned_object returned success",
+            retval);
+    }
+
 Done:
     EBPF_OBJECT_RELEASE_REFERENCE((ebpf_core_object_t*)object);
-    EBPF_RETURN_RESULT(retval);
```

Comment:
This change looks wrong to me.   EBPF_RETURN_RESULT would also log errors.  With this change, no errors will be logged, and EBPF_LOG_ENTRY would log entry but have no corresponding exit log.

Reply:
1. Upon success, this function does print success trace.
2. As stated in the description,
  Upon error, the callers will print the error trace:
_ebpf_native_reuse_map()
_ebpf_core_protocol_get_pinned_object()

Reply:
2. As noted in my comment, that violates the invariant that log entry should be paired with a log exit/return.  And I disagree that it would end up with a misleading error trace if you instead generate a success trace.

Reply:
@dthaler : Could you please help clarify your suggestion?

In ebpf_core_get_pinned_object(),
1. Re-add EBPF_RETURN_RESULT(retval);
Or,
2. Remove EBFP_LOG_ENTRY()?

Reply:
I have no preference, only that the invariant is followed.  Others may have a stronger opinion on whether to trace or not.